### PR TITLE
Feature/ngmc 250/new cus

### PR DESCRIPTION
**Pull request by onojaonoja2**
New Customer Functionality Added

### CODE REVIEW

File: app/Http/Controllers/CustomerController.php [hunk 1]
```diff
@@ -2,9 +2,17 @@
...
 
-use Illuminate\Http\Request;
+use App\Http\Requests\CreateCustomerRequest;
+use App\Jobs\CreateCustomerJob;
+use Illuminate\Http\RedirectResponse;
 
 class CustomerController extends Controller
 {
-    //
+    public function store(CreateCustomerRequest $request): RedirectResponse
```
Comment (Skillz-systems) [inline app/Http/Controllers/CustomerController.php:11]: every thing is wrong about this method, this task should not even be a controller , it should be a job, that would be registered to tags, and the code is not good either, and even if this was a controller task, no swagger documentation

File: app/Jobs/CreateCustomerJob.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        //
+        $this->data = $data;
+    }
+
+    /**
+     * Execute the job.
+     */
+    public function handle()
+    {
+        // Create a new customer using the provided data
```
Comment (Skillz-systems) [inline app/Jobs/CreateCustomerJob.php:32]: the aim of jobs is to push created data to other services, please look at other publishing job and how they are used, and when dispatching we point it to a particular , we usually do that via a look based on the env and nnpcreusable service config

File: app/Models/Customer.php [hunk 3]
```diff
@@ -8,4 +8,13 @@
 class Customer extends Model
```
Comment (Skillz-systems) [inline app/Models/Customer.php:8]: this model should not extend model , it should extend Authenticatable

File: app/Providers/BroadcastServiceProvider.php [hunk 4]
```diff
@@ -14,6 +14,6 @@ public function boot(): void
     {
         Broadcast::routes();
```
Comment (Skillz-systems) [inline app/Providers/BroadcastServiceProvider.php:15]: why was there any modification done to this file ?

File: composer.json [hunk 5]
```diff
@@ -65,7 +66,8 @@
         "sort-packages": true,
         "allow-plugins": {
             "pestphp/pest-plugin": true,
-            "php-http/discovery": true
+            "php-http/discovery": true,
+            "wikimedia/composer-merge-plugin": true
```
Comment (Skillz-systems) [inline composer.json:70]: why was this line added

File: composer.json [hunk 6]
```diff
@@ -13,6 +13,7 @@
         "laravel/sanctum": "^3.3",
         "laravel/tinker": "^2.8",
         "league/flysystem-aws-s3-v3": "^3.0",
+
```
Comment (Skillz-systems) [inline composer.json:16]: why this space here

File: config/queue.php [hunk 7]
```diff
@@ -76,7 +76,7 @@
             'hosts' => [
                 [
                     'host' => env('RABBITMQ_HOST', ''),
-                    'port' => env('RABBITMQ_PORT', 5672),
+                    'port' => (int) env('RABBITMQ_PORT', 5672),
```
Comment (Skillz-systems) [inline config/queue.php:79]: why do we have this modification

File: tests/Feature/CreateCustomerTest.php [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+            'email' => 'test@example.com',
+            'phone_number' => '1234567890',
+            'password' => bcrypt('password'),
+            'created_by_user_id' => 1,
+            'status' => 1,
+        ];
+
+        // Dispatch the job with the customer data
+        $this->artisan('queue:work --once');
+        dispatch(new \App\Jobs\CreateCustomerJob($data));
```
Comment (Skillz-systems) [inline tests/Feature/CreateCustomerTest.php:27]: this test is completely wrong

**Comment by onojaonoja2** [comment]
Okay, redoing the task and implementing the feedback.

**Review by Skillz-systems — APPROVED** [review]
